### PR TITLE
Questions controller

### PR DESCRIPTION
**Pull request by KhoiLe89**
Completed Edit/Delete functionality for Questions Controller, as well as Unit tests for said methods.

Aside from the Questions Controller:

-Fixed some styling in regards to the delete button for surveys. 
-Created question factory

### CODE REVIEW

File: app/controllers/questions_controller.rb [hunk 1]
```diff
@@ -21,6 +21,34 @@ def show
...
+      redirect_to presentation_survey_path(@presentation.id, @survey.id)
+    else
+      render :edit
+    end
+  end
+
+  def destroy
+    @presentation = Presentation.find(params[:presentation_id])
+    @survey = Survey.find(params[:survey_id])
+    @question = @survey.questions.find(params[:id])
```
Comment (gxespino) [inline app/controllers/questions_controller.rb]: `#edit`, `#update`, `#destroy` all have the same instance_variables being called. Can we DRY it up somehow?

File: app/views/surveys/show.html.erb [hunk 2]
```diff
@@ -11,7 +11,10 @@
     <ul>
       <li><%= q.order %>
       <%= q.prompt %>
-      <%= q.response_type %></li>
+      <%= q.response_type %>
```
Comment (gxespino) [inline app/views/surveys/show.html.erb]: Lets use variable names that make sense (e.g. `question` vs `q`) across the app. It's more typing now but in 6 months when you or someone else has to maintain this, 1 letter variable names add an unnecessary layer of complexity.

File: test/controllers/questions_controller_test.rb [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  describe '#create' do
+    it "Should create a new question if User is an Admin" do
+      admin = create :user, :admin
+      presentation = create :presentation
+      survey = create :survey, order: 1, subject: "Git", presentation_id: presentation.id
+      question = create :question
+
+      sign_in admin
+
+      post :create, params: {presentation_id: presentation.id, survey_id: survey.id, question:{order: question.order, prompt: question.prompt, response_type: question.response_type}}
```
Comment (gxespino) [inline test/controllers/questions_controller_test.rb]: Ruby style guides recommend spaces around brackets. see: [ruby style guide](https://github.com/bbatsov/ruby-style-guide#spaces-braces)

File: test/controllers/questions_controller_test.rb [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+      assert_redirected_to presentation_survey_path(presentation.id, survey.id), "Create method unsuccessful, no redirect to presentation_survey_path"
```
Comment (gxespino) [inline test/controllers/questions_controller_test.rb]: Separate assertions from actions in tests.

**Comment by gxespino** [comment]
@KhoiLe89: No, it's fine as it was before. I don't know what your concerns are doing but it's generally not the Rails pattern. I was talking mainly about maybe extracting a method such as:

```ruby
def edit
  set_instance_variables
  ...
end

def update
  set_instance_variables
  ...
end

def destroy
  set_instance_variables
  ...
end

private

def set_instance_variables
  @presentation = Presentation.find(params[:presentation_id])
  @survey = Survey.find(params[:survey_id])
  @question = @survey.questions.find(params[:id])
end
```

**Comment by KhoiLe89** [comment]
@gxespino Oh of course! Question though, could you potentially DRY up controllers using this method? I was working off this here http://cloudspace.com/blog/2015/10/12/dry-rails-controllers#.WLbogrYrLVp.

**Comment by gxespino** [comment]
You could do it like that blog, but I have no idea what it's doing without sitting down and trying to understand it. Explicitness is better at this stage of the game, I think.

**Pull request merged**